src/main/java/ch/mfrey/jackson/antpathfilter/AntPathPropertyFilter.java

```diff
@@ -107,40 +118,90 @@ protected boolean include(final PropertyWriter writer) {
     /**
      * Include.
      * 
-     * @param writer
-     *            the writer
-     * @param jgen
-     *            the jgen
+     * @param writer the writer
+     * @param jgen the jgen
      * @return true, if successful
      */
     protected boolean include(final PropertyWriter writer, final JsonGenerator jgen) {
         String pathToTest = getPathToTest(writer, jgen);
         if (_propertiesToInclude.isEmpty()) {
+
```

Comment:
In fact I think we can further optimize this cache implementation by completely avoiding the the "for loops". I wonder if you like to do it yourself and if you have time doing it. 
If you like to do it it's pretty simple:
- after `pathToTest` is known check directly the cache
- if cache found return result (true, false)
- else do the standard calculation + adding the result into the cache just before returning it.

Like this we can avoid all the for loops